test/integration/api/user-preferences.js

```diff
@@ -0,0 +1,172 @@
+const { testService } = require('../setup');
+
+describe('api: user-preferences', () => {
+
```

Comment:
I was playing around with this new endpoint (with a notebook and pyodk) and saw that i could make properties named `  ` and ` whitespace ` and so on. I think the property values being pretty unrestricted is fine but having `prop` be different from ` prop   ` seems potentially problematic. 

Would love to see a test showing propertyNames get trimmed.

Reply:
If we were talking XForm spec (Forms, Submissions, Datasets, Entities), we'd want to (and do) get very clear about what counts as the same or different values, because we want to be as clear and as _nice_ to form designers as possible and not have something not work as expected because the casing or whitespace was subtly different.

But you're right that in this context, we've talked about how this user preferences endpoint is mainly for Frontend to use, so it makes sense to keep it simple and use the property name as is.